### PR TITLE
[TECH] Adding internationalization

### PR DESCRIPTION
**Pull request by tensenrey**

### CODE REVIEW

**Review by fshovchko — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,8 @@ package-lock.json
 
 # production
 /build
+src/locales/en/messages.js
+src/locales/ru/messages.js
```
Comment (mtalstykh-exadel) [inline .gitignore]: src/locales/*/messages.js

File: src/components/buttons/Buttons.jsx [hunk 2]
```diff
@@ -19,19 +20,27 @@ const Buttons = () => {
...
+    <>
+    <span>Current locale: {i18n.locale}</span>
       <Button
         variant="contained"
         color="primary"
         disableElevation
         className={classes.testButtons}
       >
-        Example button Material-UI
+        <Trans id="Example button Material-UI">Example button Material-UI</Trans>
```
Comment (mtalstykh-exadel) [inline src/components/buttons/Buttons.jsx]: text inside <Trans> tag is already an ID.  So you can do `<Trans>Example button Material-UI</Trans>`

File: src/index.js [hunk 3]
```diff
@@ -3,14 +3,23 @@ import ReactDOM from 'react-dom';
...
+import { i18n } from '@lingui/core';
+import { I18nProvider } from '@lingui/react';
 import { BrowserRouter } from 'react-router-dom';
 import { Provider } from 'react-redux';
 import store from './store/store';
+import catalogEn from '@lingui/loader!./locales/en/messages.po';
+import catalogRu from '@lingui/loader!./locales/ru/messages.po';
+i18n.load('en', catalogEn.messages);
+i18n.load('ru', catalogRu.messages);
+i18n.activate('en');
```
Comment (mtalstykh-exadel) [inline src/index.js]: make a dynamic loading based on localstorage. You can create a langService or king of and move all logic there

**Review by fshovchko — DISMISSED** [review]

File: src/utils/lang-service.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    switch (lang) {
+      case "ru":
+        i18n.load("ru", catalogRu.messages);
+        i18n.activate("ru");
+        break;
+      case "en":
+        i18n.load("en", catalogEn.messages);
+        i18n.activate("en");
+        break;
+    }
```
Comment (mtalstykh-exadel) [inline src/utils/lang-service.js]: you forgot about 'default'

File: src/utils/lang-service.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { i18n } from "@lingui/core";
+import catalogEn from "@lingui/loader!../locales/en/messages.po";
+import catalogRu from "@lingui/loader!../locales/ru/messages.po";
+
+class Lang {
```
Comment (mtalstykh-exadel) [inline src/utils/lang-service.js]: use named export

File: src/components/buttons/Buttons.jsx [hunk 6]
```diff
@@ -19,19 +20,24 @@ const Buttons = () => {
...
         color="primary"
         disableElevation
         className={classes.testButtons}
       >
-        Example button Material-UI
+        <Trans>Русский</Trans>
       </Button>
       <Button variant="contained" disabled className={classes.testButtons}>
-        Example №2 disabled
+        <Trans>Английский</Trans>
```
Comment (mtalstykh-exadel) [inline src/components/buttons/Buttons.jsx]: why russian by default? Use ENG

File: .linguirc [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+   "locales": ["en", "ru"],
+   "catalogs": [{
+      "path": "src/locales/{locale}/messages",
+      "include": ["src"]
+   }],
+   "format": "po"
```
Comment (mtalstykh-exadel) [inline .linguirc]: use a sourceLocale='en' option

File: src/locales/en/messages.po [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+"PO-Revision-Date: \n"
+"Last-Translator: \n"
+"Language: \n"
+"Language-Team: \n"
+"Content-Type: \n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#: src/components/buttons/Buttons.jsx:34
+msgid "Английский"
```
Comment (mtalstykh-exadel) [inline src/locales/en/messages.po]: better to use english as an ID

**Review by fshovchko — DISMISSED** [review]

**Review by fshovchko — DISMISSED** [review]

**Review by mtalstykh-exadel — APPROVED** [review]

**Review by kseniyaLine — APPROVED** [review]

**Pull request merged**